pkg/apis/kops/cluster.go

```diff
@@ -199,6 +199,10 @@ type ClusterSpec struct {
 
 	// ClusterAutoscaler defines the cluster autoscaler configuration.
 	ClusterAutoscaler *ClusterAutoscalerConfig `json:"clusterAutoscaler,omitempty"`
+
+	// CSI defines container storage interface configuration.
+	// +optional
+	CSI *CSIConfiguration `json:"csi,omitempty"`
```

Comment:
For other providers we have similar types of configuration under a struct in `CloudConfiguration`. AWS doesn't have its own struct here; now may be the time to add it.

Reply:
I thought about that, but CloudConfig seems to be tightly mapped to the actual config file that is placed on disk, which is deprecated according to documentation I found.  Shouldn't we move on from that object?  Also, it kindof combines/conflates various addons, like CCM, CSI, etc. (even though, yes they are cloud related, there are many combinations that users might want to use and combining all of the configuration makes it quite difficult to understand what configuration is configuring which addon).